tenjin/actions/bing_search.py

```diff
@@ -0,0 +1,28 @@
+from typing import List
+from langchain.utilities import BingSearchAPIWrapper
+
+class BingSearch(BingSearchAPIWrapper):
+    """
+    Wrapper around Bing Search API that returns a list of results, sources, and metadata.
+    """
+
+    name = "Bing Search"
+    description = "Useful if you need to answer questions about current events. You should ask targeted questions."
+
+    def sources(self, query) -> List[dict]:
+        sources = []
+        results = super().results(query, 5)
+
+        for result in results:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To create a `list`, try to use `list` comprehension instead of a loop. List comprehension is the preferred way to make a list using Python, and it&#39;s simpler and easier to understand than using a loop.

[Learn more](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)